ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-add/index.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import styled from '../../styles/styled-components'
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+import * as PopoutView from '../../../component/dropdown/popout/dropdown.popout.view'
+import * as ListCreateView from '../../../component/list-create/list-create.view'
+
+type Result = {
+  matchesFilter?: boolean
+  alreadyContains?: boolean
+  icon: any
+  title: string
+  id: any
+}
+
+type Props = {
+  items?: [Result]
+  model: any
+  bookmarkHandler: (id: any) => void
+}
+
+const Root = styled.div`
+  padding: ${props => props.theme.minimumSpacing};
+`
+
+const Button = styled.button`
+  width: 100%;
+  display: block;
+  padding: 0px ${props => props.theme.largeSpacing};
+  text-align: left;
+  height: ${props => props.theme.minimumButtonSize};
+  line-height: ${props => props.theme.minimumButtonSize};
+
+  .composed-button {
+    width: 100%;
+    display: block;
+  }
+`
+
+const IconSpan = styled.span`
+  margin-right: ${props => props.theme.mediumSpacing};
+`
+
+const getIconStyle = (result: Result) => {
+  if (result.alreadyContains) {
+    return 'fa-check-square'
+  }
+
+  if (result.matchesFilter) {
+    return 'fa-square'
+  }
+
+  return 'fa-ban'
+}
+
+const renderButton = (result: Result, props: Props) => (
+  <Button
+    className={`is-button is-neutral ${!result.alreadyContains &&
+      !result.matchesFilter &&
+      'is-disabled'}`}
+    key={result.id}
+    onClick={() => props.bookmarkHandler(result.id)}
+  >
+    <IconSpan className={`fa ${getIconStyle(result)}`} />
+    <span className={result.icon} />
+    <span>{result.title}</span>
+  </Button>
+)
+
+export default hot(module)((props: Props) => (
+  <Root {...props}>
+    <div className="is-header">Add / remove from lists</div>
```

Comment:
You should bring the css into styled components instead of using class names when possible

Reply:
Agreed, but these classes are global rather than contextually close to the component

Reply:
These global mixins will eventually go away. If we catch as many as possible now and move them into the component, we can get ahead of as many of the inevitable styling bugs as possible.